Buildings/Fluid/Dehumidifiers/Desiccant/BaseClasses/Performance.mo

```diff
@@ -0,0 +1,301 @@
+    "In " + getInstanceName() + ": temperature of the process air entering the dehumidifier is beyond 
+    the range that is defined in the performance curve.",
+    level=AssertionLevel.error);
+     assert(X_w_ProEnt <= per.X_w_ProEnt_max and X_w_ProEnt >= per.X_w_ProEnt_min,
+    "In " + getInstanceName() + ": humidity ratio of the process air entering the dehumidifier is beyond 
+    the range that is defined in the performance curve.",
+    level=AssertionLevel.error);
+    VReg_flow =
+      Buildings.Fluid.Dehumidifiers.Desiccant.BaseClasses.performanceCurve(
+      TProEnt=TProEnt,
+      X_w_ProEnt=X_w_ProEnt,
+      vPro=VPro_flow/VPro_flow_nominal*vPro_nominal,
+      a=per.coevReg)/vReg_nominal*VReg_flow_nominal*uSpe;
+     assert(VReg_flow <= VReg_flow_nominal,
+     "In " + getInstanceName() + ": regeneration flow rate is not sufficient.",
+     level=AssertionLevel.error);
+     TProLea = TProEnt + (Buildings.Fluid.Dehumidifiers.Desiccant.BaseClasses.performanceCurve(
+        TProEnt = TProEnt,
+        X_w_ProEnt = X_w_ProEnt,
+        vPro = VPro_flow/VPro_flow_nominal*vPro_nominal,
+        a = per.coeTProLea)+ 273.15 - TProEnt)*etaSen;
+     X_w_ProLea = X_w_ProEnt-(X_w_ProEnt-(Buildings.Fluid.Dehumidifiers.Desiccant.BaseClasses.performanceCurve(
+        TProEnt = TProEnt,
+        X_w_ProEnt = X_w_ProEnt,
+        vPro = VPro_flow/VPro_flow_nominal*vPro_nominal,
+        a = per.coeX_w_ProLea)))*etaLat;
+     assert(X_w_ProLea > 0,
+     "In " + getInstanceName() + ": humidity ratio of the process air leaving 
+     the dehumidifier becomes negative.",
+     level=AssertionLevel.error);
+     CpReg = Buildings.Utilities.Math.Functions.smoothMax(
+        x1 = Buildings.Fluid.Dehumidifiers.Desiccant.BaseClasses.performanceCurve(
+                TProEnt = TProEnt,
+                X_w_ProEnt = X_w_ProEnt,
+                vPro = VPro_flow/VPro_flow_nominal*vPro_nominal,
+                a = per.coeQReg_flow)*uSpe,
+        x2 = 0,
+        deltaX = 0.01);
+      yQReg = CpReg*(X_w_ProEnt - X_w_ProLea)*mPro_flow*(per.TRegEnt_nominal -
+      TRegEnt)/(per.TRegEnt_nominal - TProEnt)/QReg_flow_nominal;
+    assert(yQReg <= 1,
+     "In " + getInstanceName() + ": heating power is not sufficient for regeneration.",
+     level=AssertionLevel.error);
+  else
+    //No dehumidification occurs.
+    TProLea = TProEnt;
+    X_w_ProLea = X_w_ProEnt;
+    VReg_flow = 0;
+    CpReg = 0;
+    yQReg = 0;
+    etaSen = 0;
+    etaLat = 0;
+  end if;
+  annotation (
+  defaultComponentName="dehPer",
+  Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
+        coordinateSystem(preserveAspectRatio=false)),
+    Documentation(info="<html>
+<p>
+This model calculates the outlet condition of the process air of a desiccant dehumidifier based on
+the inlet condition of the process air.
+Specifically, this calculation is configured as follows.
+</p>
+<ul>
+ <li>
+ If the dehumidification signal <code>onDeh=true</code>,
+ <ul>
+  <li>
```

Comment:
make sure no empty space in front of each sentence.

Reply:
Removed the extra spaces.